File: unified-codes/libs/data/src/lib/types/DrugInteraction.ts
export interface IDrugInteraction {
  name: string;
  description: string;
  severity: string;
  source: string;
  rxcui: string;
}

export class DrugInteraction implements IDrugInteraction {
  private _name: string;
  private _description: string;
  private _severity: string;
  private _source: string;
  private _rxcui: string;

  constructor(drugInteraction: IDrugInteraction) {
    this._name = drugInteraction.name;
    this._description = drugInteraction.description;
    this._severity = drugInteraction.severity;
    this._source = drugInteraction.source;
    this._rxcui = drugInteraction.rxcui;
  }

  get name(): string {
    return this._name;
  }

  get description(): string {
    return this._description;
  }

  get severity(): string {
    return this._severity;
  }

  get source(): string {
    return this._source;
  }

  get rxcui(): string {
    return this._rxcui;
  }
}

export default DrugInteraction;

File: backend/graphql_v1/universal_codes/src/types/drug_interaction.rs
use async_graphql::*;

#[derive(Clone, Debug)]
pub struct DrugInteractionType {
    pub description: String,
    pub name: String,
    pub rxcui: String,
    pub severity: String,
    pub source: String,
}

#[Object]
impl DrugInteractionType {
    pub async fn description(&self) -> &str {
        &self.description
    }
    pub async fn name(&self) -> &str {
        &self.name
    }
    pub async fn rxcui(&self) -> &str {
        &self.rxcui
    }
    pub async fn severity(&self) -> &str {
        &self.severity
    }
    pub async fn source(&self) -> &str {
        &self.source
    }
}
